# Catalogues exploration

## OpenNGC catalogue
 Voir [source](https://github.com/mattiaverga/OpenNGC).

In [484]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 500)
CHEMIN_SOURCES = "/Users/lyes/Desktop/MD5/Python/python_projet/Data/données brutes/"

In [124]:
col_names = ['NGC_IC_designation', 
    'object_type',
    'Right_Ascension',
    'Declinaison',
    'Constelation_name_abrv',
    'major_axis',
    'minor_axis',
    'major__axis_position_angle', 
    'B_Apparent_Magnitude',
    'V_Apparent_Magnitude',             
    'J_Apparent_Magnitude',
    'H_Apparent_Magnitude',
    'K_Apparent_Magnitude',
    'Mean_surface_brigthness', 
    'Hubble_morphological_type',
    'center_star_U_maghitude',
    'center_star_B_maghitude',
    'center_star_V_maghitude',
    'Messier_number',
    'NGC_numner',
    'IC_number',
    'Center_star_name',
    'Identifiers',
    'Common_names',       
    'NED_notes',  
    'OpenNGC_notes']

NGC_mattiaAverga_GitHub = pd.read_csv(CHEMIN_SOURCES+"NGC_mattiaAverga_GitHub.csv", sep=';',header=0, names=col_names)
NGC_mattiaAverga_GitHub.shape

(13958, 26)

### Adendum

In [125]:
addendum_NGC_mattiaAverga_GitHub = pd.read_csv(CHEMIN_SOURCES+"addendum_NGC_mattiaAverga_GitHub.csv", sep=';', header=0, names=col_names)
addendum_NGC_mattiaAverga_GitHub.shape

(20, 26)

## Messier Catalogue Construction


### Messier catalogue of Datastro
[link](https://www.datastro.eu/explore/dataset/catalogue-de-messier/table/?disjunctive.objet&disjunctive.mag&disjunctive.english_name_nom_en_anglais&disjunctive.french_name_nom_francais&disjunctive.latin_name_nom_latin&sort=messier)

In [227]:
col_names = ['Messier_number',
    'NGC_IC_designation', 
    'object_type',
    'Season',
    'Magnitude',
    'Constellation_EN',
    'Constellation_FR',
    'Constellation_Latin',
    'Right_Ascension',        
    'Declinaison',
    'Distance_light_year',
    'Size',
    'Discoverer',
    'Year',
    'Image1',
    'Image2',
    'Constelation_abr' ]
messier_Datastro_website = pd.read_csv(CHEMIN_SOURCES+"messier_Datastro_website.csv", sep=';', header=0, names=col_names)
messier_Datastro_website.shape

(110, 17)

In [297]:
# Corrections : 
messier_Datastro_website.loc[messier_Datastro_website['NGC_IC_designation']=='NGC 9176', 'NGC_IC_designation'] = 'NGC 1976'

In [228]:
NGC_mattiaAverga_GitHub.loc[(NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4725') |
                        (NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4715')]

NGC_IC_designation object_type Right_Ascension  Declinaison  \
4923             IC4715        *Ass     18:16:56.12  -18:30:52.4   
4933             IC4725         OCl     18:31:46.77  -19:06:53.8   

     Constelation_name_abrv  major_axis  minor_axis  \
4923                    Sgr       120.0        60.0   
4933                    Sgr        14.1         NaN   

      major__axis_position_angle  B_Apparent_Magnitude  V_Apparent_Magnitude  \
4923                         NaN                   NaN                   4.5   
4933                         NaN                  5.29                   4.6   

      J_Apparent_Magnitude  H_Apparent_Magnitude  K_Apparent_Magnitude  \
4923                   NaN                   NaN                   NaN   
4933                   NaN                   NaN                   NaN   

      Mean_surface_brigthness Hubble_morphological_type  \
4923                      NaN                       NaN   
4933                      NaN                       NaN   

      center_star_U_maghitude  center_star_B_maghitude  \
4923                      NaN                      NaN   
4933                      NaN                      NaN   

      center_star_V_maghitude  Messier_number NGC_numner IC_number  \
4923                      NaN            24.0        NaN       NaN   
4933                      NaN            25.0        NaN       NaN   

     Center_star_name Identifiers          Common_names  \
4923              NaN         NaN  Small Sgr Star Cloud   
4933              NaN         NaN                   NaN   

                  NED_notes                             OpenNGC_notes  
4923  Milky Way star cloud.  V-mag taken from HEASARC's messier table  
4933                    NaN                                       NaN

According to this [link](http://messier.obspm.fr/m-q&a.html):
- M24: the Milky Way Patch in Sagittarius; it contains however the 11th magnitude cluster NGC 6603 which is sometimes erroneously listed as M24. Also, it may be that IC 4715 is M24. 
- M25: this is IC 4725, 
- M40: the double star Winnecke 4
- M45: the Pleiades; however, this cluster is associated with nebulae which have NGC numbers.

*Note:* M24 and M25 exist in NGC_mattiaAverga_GitHub. But M40 and M45 does not exist. So, we need to separate of from the dataset of Messier catalogue the objects M40 and M45. Then, merge the M40 and M45 with NGC Addendum dataset (which contains these two objects)

#### Spliting of Messier dataset

In [298]:
messier_Datastro_website_without_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']!='M40') &
                        (messier_Datastro_website['Messier_number']!='M45')].copy()
messier_Datastro_website_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']=='M40') |
                        (messier_Datastro_website['Messier_number']=='M45')].copy()

#### Merging of Messier and NGC datasets

In [299]:
# Normalise the NGC_designation: NGCnnnn
#1
messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==8
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "")

messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==7
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "0")

# Merge of first dataset
merge_messier_Datastro_website_without_M40_M45 = messier_Datastro_website_without_M40_M45.merge(NGC_mattiaAverga_GitHub, how='left', on='NGC_IC_designation')
merge_messier_Datastro_website_without_M40_M45.shape

(108, 42)

Messier_number_x
NGC_IC_designation
*object_type_x* (delete object_type_y is abreviation) 
Season                         
Magnitude                       
Constellation_EN               
Constellation_FR               
Constellation_Latin            
Right_Ascension_x  1 Nan(keep Right_Ascension_y no nan)            
Declinaison_x  1 Nan(keep Declinaison_y no nan)                
Distance_light_year           
Size                           
Discoverer                     
Year                          
Image1                         
Image2                         
*Constelation_abr (Delete)*             
*object_type_y  (delete)*            
*Right_Ascension_y*             
*Declinaison_y*                  
Constelation_name_abrv      
major_axis                   
minor_axis                    
major__axis_position_angle    
B_Apparent_Magnitude          
V_Apparent_Magnitude          
J_Apparent_Magnitude          
H_Apparent_Magnitude          
K_Apparent_Magnitude          
Mean_surface_brigthness       
Hubble_morphological_type      
center_star_U_maghitude       
center_star_B_maghitude       
center_star_V_maghitude       
*Messier_number_y (delete)*             
NGC_numner  (6 double-listed in NGC)                  
*IC_number (delete: 108 Nan)*                    
Center_star_name               
Identifiers                    
Common_names                   
NED_notes                      
OpenNGC_notes

In [363]:
drop_list = ['object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'Messier_number_y', 'IC_number']
rename_lis = {'Messier_number_x' : 'Messier_number',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_without_M40_M45.shape

(108, 37)

In [317]:
# Merge of second dataset (M40 and M45)
messier_Datastro_website_M40_M45

Messier_number NGC_IC_designation                  object_type  \
55            M45                NaN   Open Cluster / Amas Ouvert   
97            M40         Winnecke 4  Double star / Étoile Double   

                Season  Magnitude Constellation_EN Constellation_FR  \
55      Winter / Hiver          1              NaN              NaN   
97  Spring / Printemps          9              NaN              NaN   

   Constellation_Latin Right_Ascension Declinaison  Distance_light_year  \
55                 NaN             NaN         NaN                410.0   
97                 NaN             NaN         NaN                  NaN   

      Size Discoverer    Year                                Image1  \
55  120,0'        NaN     NaN  http://www.lasam.ca/messier/M045.JPG   
97     NaN   Hevelius  1660.0  http://www.lasam.ca/messier/M040.JPG   

                                               Image2 Constelation_abr  
55  https://www.datastro.eu/api/v2/catalog/dataset...              NaN  
97  https://www.datastro.eu/api/v2/catalog/dataset...              NaN

In [352]:
Open_NGC_M40_M45 = addendum_NGC_mattiaAverga_GitHub[~addendum_NGC_mattiaAverga_GitHub['Messier_number'].isna()].copy()
Open_NGC_M40_M45['Messier_number_bis'] = 'M'+Open_NGC_M40_M45['Messier_number'].astype('int').astype('str')
Open_NGC_M40_M45 = Open_NGC_M40_M45.drop(columns='Messier_number')
Open_NGC_M40_M45 = Open_NGC_M40_M45.rename(columns={'Messier_number_bis': 'Messier_number'})
Open_NGC_M40_M45

NGC_IC_designation object_type Right_Ascension Declinaison  \
13               M040          **      12:22:16.1   +58:05:04   
14             Mel022         OCl      03:47:28.6   +24:06:19   

   Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
13                    UMa         NaN         NaN                         NaN   
14                    Tau       150.0       150.0                        90.0   

    B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
13                   NaN                   8.0                   NaN   
14                   NaN                   1.2                   NaN   

    H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
13                   NaN                   NaN                      NaN   
14                   NaN                   NaN                      NaN   

   Hubble_morphological_type  center_star_U_maghitude  \
13                       NaN                      NaN   
14                       NaN                      NaN   

    center_star_B_maghitude  center_star_V_maghitude  NGC_numner  IC_number  \
13                      NaN                      NaN         NaN        NaN   
14                      NaN                      NaN         NaN        NaN   

    Center_star_name        Identifiers Common_names NED_notes  \
13               NaN  WDS J12222+5805AB          NaN       NaN   
14               NaN          MWSC 0305     Pleiades       NaN   

                                        OpenNGC_notes Messier_number  
13            VMag taken from HEASARC's messier table            M40  
14  Coordinates taken from Simbad, VMag taken from...            M45

In [364]:
merge_messier_Datastro_website_M40_M45 = messier_Datastro_website_M40_M45.merge(Open_NGC_M40_M45, how='left', on='Messier_number')
merge_messier_Datastro_website_M40_M45

Messier_number NGC_IC_designation_x                object_type_x  \
0            M45                  NaN   Open Cluster / Amas Ouvert   
1            M40           Winnecke 4  Double star / Étoile Double   

               Season  Magnitude Constellation_EN Constellation_FR  \
0      Winter / Hiver          1              NaN              NaN   
1  Spring / Printemps          9              NaN              NaN   

  Constellation_Latin Right_Ascension_x Declinaison_x  Distance_light_year  \
0                 NaN               NaN           NaN                410.0   
1                 NaN               NaN           NaN                  NaN   

     Size Discoverer    Year                                Image1  \
0  120,0'        NaN     NaN  http://www.lasam.ca/messier/M045.JPG   
1     NaN   Hevelius  1660.0  http://www.lasam.ca/messier/M040.JPG   

                                              Image2 Constelation_abr  \
0  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   
1  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   

  NGC_IC_designation_y object_type_y Right_Ascension_y Declinaison_y  \
0               Mel022           OCl        03:47:28.6     +24:06:19   
1                 M040            **        12:22:16.1     +58:05:04   

  Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
0                    Tau       150.0       150.0                        90.0   
1                    UMa         NaN         NaN                         NaN   

   B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
0                   NaN                   1.2                   NaN   
1                   NaN                   8.0                   NaN   

   H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
0                   NaN                   NaN                      NaN   
1                   NaN                   NaN                      NaN   

  Hubble_morphological_type  center_star_U_maghitude  center_star_B_maghitude  \
0                       NaN                      NaN                      NaN   
1                       NaN                      NaN                      NaN   

   center_star_V_maghitude  NGC_numner  IC_number  Center_star_name  \
0                      NaN         NaN        NaN               NaN   
1                      NaN         NaN        NaN               NaN   

         Identifiers Common_names NED_notes  \
0          MWSC 0305     Pleiades       NaN   
1  WDS J12222+5805AB          NaN       NaN   

                                       OpenNGC_notes  
0  Coordinates taken from Simbad, VMag taken from...  
1            VMag taken from HEASARC's messier table

In [371]:
# Move data before deleting NGC_IC_designation_y
merge_messier_Datastro_website_M40_M45.loc[merge_messier_Datastro_website_M40_M45['NGC_IC_designation_x'].isna(),
                                          'NGC_IC_designation_x'] = merge_messier_Datastro_website_M40_M45['NGC_IC_designation_y']

In [372]:
drop_list = ['NGC_IC_designation_y', 'object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'IC_number']
rename_lis = {'NGC_IC_designation_x': 'NGC_IC_designation',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_M40_M45.shape

(2, 37)

In [392]:
messier_catalogue = pd.concat([merge_messier_Datastro_website_without_M40_M45, merge_messier_Datastro_website_M40_M45])
#messier_catalogue.sort_values(by='Messier_number')

In [393]:
messier_catalogue[messier_catalogue['Hubble_morphological_type'].isna()]

### Messier Catalogue of jbcurtin_gitHub
https://github.com/jbcurtin/messier-catalogue

In [495]:
col_names = ['Messier_number_jbcurtin_gitHub',
    'NGC_IC_designation_jbcurtin_gitHub', 
    'Common_name_jbcurtin_gitHub',
    'image_jbcurtin_gitHub',
    'Object_type_jbcurtin_gitHub',
    'Distance_KLY_jbcurtin_gitHub',             
    'Constellation_jbcurtin_gitHub',
    'Apparent_magnitude_jbcurtin_gitHub',
    'Right_Ascension_jbcurtin_gitHub',        
    'Declinaison_jbcurtin_gitHub']
messier_with_picture_links_jbcurtin_gitHub = pd.read_csv(CHEMIN_SOURCES+"messier_with_picture_links_jbcurtin_gitHub.csv", keep_default_na=True, sep=';', header=0, names=col_names)
messier_with_picture_links_jbcurtin_gitHub.loc[messier_with_picture_links_jbcurtin_gitHub['Common_name_jbcurtin_gitHub']==' ',
                       'Common_name_jbcurtin_gitHub'] = np.nan #not a space
    
messier_with_picture_links_jbcurtin_gitHub.dtypes

Messier_number_jbcurtin_gitHub         object
NGC_IC_designation_jbcurtin_gitHub     object
Common_name_jbcurtin_gitHub            object
image_jbcurtin_gitHub                  object
Object_type_jbcurtin_gitHub            object
Distance_KLY_jbcurtin_gitHub           object
Constellation_jbcurtin_gitHub          object
Apparent_magnitude_jbcurtin_gitHub    float64
Right_Ascension_jbcurtin_gitHub        object
Declinaison_jbcurtin_gitHub            object
dtype: object

In [536]:
messier_catalogue2 = messier_with_picture_links_jbcurtin_gitHub.merge(messier_catalogue,how='left', left_on='Messier_number_jbcurtin_gitHub', right_on='Messier_number')
messier_catalogue2.shape

(110, 47)

In [537]:
#keep 'Common_name_jbcurtin_gitHub', delete_Common_names
messier_catalogue2.loc[messier_catalogue2['Common_name_jbcurtin_gitHub'].isna(),
                   'Common_name_jbcurtin_gitHub'] = messier_catalogue2['Common_names']

# In 'Object_type_jbcurtin_gitHub', Galaxy is subtyped. So we should keep that info
# Keep 'object_type', delete 'Object_type_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['object_type'].str.contains('Galaxy'),
                   'object_type'] = messier_catalogue2['Object_type_jbcurtin_gitHub']

#Delete 'Constellation_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['Constellation_Latin'].isna(),
                   'Constellation_Latin'] = messier_catalogue2['Constellation_jbcurtin_gitHub']

drop_list = ['Common_names', 'Messier_number_jbcurtin_gitHub', 'NGC_IC_designation_jbcurtin_gitHub', 
             'Object_type_jbcurtin_gitHub', 'Distance_KLY_jbcurtin_gitHub', 'Constellation_jbcurtin_gitHub',
            'Right_Ascension_jbcurtin_gitHub', 'Declinaison_jbcurtin_gitHub']
rename_lis = {'Common_name_jbcurtin_gitHub': 'Common_name'}

messier_catalogue2 = messier_catalogue2.drop(columns=drop_list)
messier_catalogue2 = messier_catalogue2.rename(columns=rename_lis)
messier_catalogue2.shape


(110, 39)

### Messier Catalogue of Nexstarsite
https://www.nexstarsite.com/Book/DSO.htm

In [539]:
MessierObjects_nexstarsite = pd.read_excel(CHEMIN_SOURCES+"nexstarsite/"+"MessierObjects_nexstarsite.xls", skiprows=2)
MessierObjects_nexstarsite.dtypes

ObjectNum          int64
Name              object
Type              object
Constellation     object
RAHour             int64
RAMinute         float64
DecSign           object
DecDeg             int64
DecMinute        float64
Magnitude         object
Info              object
Distance (ly)      int64
dtype: object

There is not any new data in this dataset !

In [546]:
messier_catalogue_final = messier_catalogue2

## Caldwell Catalogue

In [547]:
messier_catalogue_final.dtypes

Common_name                            object
image_jbcurtin_gitHub                  object
Apparent_magnitude_jbcurtin_gitHub    float64
Messier_number                         object
NGC_IC_designation                     object
object_type                            object
Season                                 object
Magnitude                               int64
Constellation_EN                       object
Constellation_FR                       object
Constellation_Latin                    object
Distance_light_year                   float64
Size                                   object
Discoverer                             object
Year                                  float64
Image1                                 object
Image2                                 object
Constelation_abr                       object
Right_Ascension                        object
Declinaison                            object
Constelation_name_abrv                 object
major_axis                        

In [85]:
CaldwellObjects_nexstarsite = pd.read_excel(CHEMIN_SOURCES+"nexstarsite/"+"CaldwellObjects_nexstarsite.xls", skiprows=2)
CaldwellObjects_nexstarsite.shape

(109, 11)

ObjectNum         Name               Type Constellation Magnitude  \
0            1      NGC 188       Open Cluster           Cep       8.1   
1            2       NGC 40   Planetary Nebula           Cep      12.4   
2            3     NGC 4236      Spiral Galaxy           Dra       9.7   
3            4     NGC 7023      Bright Nebula           Cep        --   
4            5       IC 342      Spiral Galaxy           Cam       9.2   
5            6     NGC 6543   Planetary Nebula           Dra       8.1   
6            7     NGC 2403      Spiral Galaxy           Cam       8.4   
7            8      NGC 559       Open Cluster           Cas       9.5   
8            9      Sh2-155      Bright Nebula           Cep        --   
9           10      NGC 663       Open Cluster           Cas       7.1   
10          11     NGC 7635      Bright Nebula           Cas        --   
11          12     NGC 6946      Spiral Galaxy           Cep       8.9   
12          13      NGC 457       Open Cluster           Cas       6.4   
13          14  NGC 869/884       Open Cluster           Per       4.3   
14          15     NGC 6826   Planetary Nebula           Cyg       8.8   
15          16     NGC 7243       Open Cluster           Lac       6.4   
16          17      NGC 147  Elliptical Galaxy           Cas       9.3   
17          18      NGC 185  Elliptical Galaxy           Cas       9.2   
18          19      IC 5146      Bright Nebula           Cyg        --   
19          20     NGC 7000      Bright Nebula           Cyg        --   
20          21     NGC 4449   Irregular Galaxy           CVn       9.4   
21          22     NGC 7662   Planetary Nebula           And       8.3   
22          23      NGC 891      Spiral Galaxy           And       9.9   
23          24     NGC 1275   Irregular Galaxy           Per      11.6   
24          25     NGC 2419   Globular Cluster           Lyn      10.4   
25          26     NGC 4244      Spiral Galaxy           CVn      10.2   
26          27     NGC 6888      Bright Nebula           Cyg        --   
27          28      NGC 752       Open Cluster           And       5.7   
28          29     NGC 5005      Spiral Galaxy           CVn       9.8   
29          30     NGC 7331      Spiral Galaxy           Peg       9.5   
..         ...          ...                ...           ...       ...   
79          80     NGC 5139   Globular Cluster           Cen       3.6   
80          81     NGC 6352   Globular Cluster           Ara       8.1   
81          82     NGC 6193       Open Cluster           Ara       5.2   
82          83     NGC 4945      Spiral Galaxy           Cen       8.7   
83          84     NGC 5286   Globular Cluster           Cen       7.6   
84          85      IC 2391       Open Cluster           Vel       2.5   
85          86     NGC 6397   Globular Cluster           Ara       5.6   
86          87     NGC 1261   Globular Cluster           Hor       8.4   
87          88     NGC 5823       Open Cluster           Cir       7.9   
88          89     NGC 6087       Open Cluster           Nor       5.4   
89          90     NGC 2867   Planetary Nebula           Car        --   
90          91     NGC 3532       Open Cluster           Car         3   
91          92     NGC 3372      Bright Nebula           Car        --   
92          93     NGC 6752   Globular Cluster           Pav       5.4   
93          94     NGC 4755       Open Cluster           Cru       4.2   
94          95     NGC 6025       Open Cluster           TrA       5.1   
95          96     NGC 2516       Open Cluster           Car       3.8   
96          97     NGC 3766       Open Cluster           Cen       5.3   
97          98     NGC 4609       Open Cluster           Cru       6.9   
98          99         none        Dark Nebula           Cru        --   
99         100      IC 2944       Open Cluster           Cen       4.5   
100        101     NGC 6744      Spiral Galaxy           Pav       8.3   
101      

## Herschel400 Catalogue

In [87]:
Herschel400_nexstarsite = pd.read_excel(CHEMIN_SOURCES+"nexstarsite/"+"Herschel400_nexstarsite.xls", skiprows=2)
Herschel400_nexstarsite.shape

(400, 11)

# Catalogues Merging